### PR TITLE
add new issue template

### PR DESCRIPTION
**Pull request by unixfox**
## What does this PR do?

This PR is for adding a new issue template.

## Why is this change important?

This help maintainers and the community for resolving issues. 

## How to test this PR locally?

This requires to create a blank GitHub repository for testing this PR.

## Author's checklist

N/A

## Related issues

N/A

### CODE REVIEW

**Comment by dalf** [comment]
Which path should take an user of a public instance encountering a bug ?

**Comment by unixfox** [comment]
> Which path should take an user of a public instance encountering a bug ?

Well there are two possible ways:

- The first one: Like we always do, ask to contact the maintainer of the public instance to solve the bug. So we would include in the issue template to avoid reporting bug from public instances and explain how to try to contact a maintainer. Also, to use another instance if an instance doesn't work.
- The second one: Have a section for that on GitHub discussions

With the second one, our GitHub discussions may be spammed by discussions about bugs on public instances. And with the first one, we will have near zero issues/discussions about that.

I don't really know what's the best of the two ones but I tend to prefer the first one.
Until #1972 get solved the users will still try to report bugs here from public instances because they don't know how to contact a maintainer of a public instance. Maybe we could write a wiki page for helping about that: check if the instance was listed as an issue on searx-instances, use whois, ...

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+name: Bug report
+about: Report a bug in Searx
+title: ''
+labels: bug
+assignees: ''
+
+---
+**Version of Searx and commit number if you are using on master branch**
```
Comment (dalf) [inline .github/ISSUE_TEMPLATE/bug-report.md]: Looking at searx.space some instances say they are running searx version 0.16.0 and at the same time the google engine is working. It should not be the case, see https://github.com/searx/searx/pull/2236

My guess is there are a lot of custom versions / forks.

**Comment by dalf** [comment]
> With the second one, our GitHub discussions may be spammed by discussions about bugs on public instances.

This is what we have now with the issues. In this scenario, a template to ask the instance URL is helpful.

> And with the first one, we will have near zero issues/discussions about that.

In this scenario:
* the admin needs to modify the links at the bottom of the pages : https://github.com/searx/searx/blob/93c2603561c039fb43137c251493e77032f91743/searx/templates/oscar/base.html#L89-L90
* so actually : https://searx.github.io/searx/dev/makefile.html?highlight=brand#id7

A possible use case:
1. the admin forks searx (on github or elsewhere).
1. the admin modifies GIT_URL, SEARX_URL and ISSUE_URL (so there is a dedicated issue tracker / tool for this instance).
1. the admin registers the instance and acknowledge to take care of the user issues.
1. the searx team adds the instance
1. an user creates an issue on the dedicated issue tracker.
1. the admin checks whether it is a configuration problem or if it is a searx bug.
1. if this is a searx bug, the admin creates an issue on the searx issue tracker with a possible reference to the user issue.

I'm don't know if the user will get lost in the process.

Notes:
* spot.ecloud.global instance has a dedicated repository: https://gitlab.e.foundation/e/cloud/my-spot but this repository doesn't allow to create an issue.
* in the [debian package](https://salsa.debian.org/debian/searx), the issues are managed in a dedicated tool: https://tracker.debian.org/pkg/searx
* [Yunohost integration](https://github.com/YunoHost-Apps/searx_ynh/issues) is github repository

**Comment by dalf** [comment]
I have written a lot of words....

But we can start simple, as you said:

> Have a section for that on GitHub discussions

File: .github/ISSUE_TEMPLATE/bug-report.md (hunk 1 above)
Comment (unixfox) [reply .github/ISSUE_TEMPLATE/bug-report.md]: Ok I'll include a notice if one have a fork and should stipulate it.

**Comment by unixfox** [comment]
> > With the second one, our GitHub discussions may be spammed by discussions about bugs on public instances.
> 
> This is what we have now with the issues. In this scenario, a template to ask the instance URL is helpful.
> 
> > And with the first one, we will have near zero issues/discussions about that.
> 
> In this scenario:
> 
>     * the admin needs to modify the links at the bottom of the pages : https://github.com/searx/searx/blob/93c2603561c039fb43137c251493e77032f91743/searx/templates/oscar/base.html#L89-L90
> 
>     * so actually : [searx.github.io/searx/dev/makefile.html?highlight=brand#id7](https://searx.github.io/searx/dev/makefile.html?highlight=brand#id7)
> 
> 
> A possible use case:
> 
>     1. the admin forks searx (on github or elsewhere).
> 
>     2. the admin modifies GIT_URL, SEARX_URL and ISSUE_URL (so there is a dedicated issue tracker / tool for this instance).
> 
>     3. the admin registers the instance and acknowledge to take care of the user issues.
> 
>     4. the searx team adds the instance
> 
>     5. an user creates an issue on the dedicated issue tracker.
> 
>     6. the admin checks whether it is a configuration problem or if it is a searx bug.
> 
>     7. if this is a searx bug, the admin creates an issue on the searx issue tracker with a possible reference to the user issue.
> 
> 
> I'm don't know if the user will get lost in the process.
> 
> Notes:
> 
>     * spot.ecloud.global instance has a dedicated repository: [gitlab.e.foundation/e/cloud/my-spot](https://gitlab.e.foundation/e/cloud/my-spot) but this repository doesn't allow to create an issue.
> 
>     * in the [debian package](https://salsa.debian.org/debian/searx), the issues are managed in a dedicated tool: [tracker.debian.org/pkg/searx](https://tracker.debian.org/pkg/searx)
> 
>     * [Yunohost integration](https://github.com/YunoHost-Apps/searx_ynh/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc) is github repository

Well it seems a bit of complicated setup.

If we keep the process simple: start with a GitHub discussions section for that, then when a correct way for contacting an instance maintainer will be established we can close the GitHub discussions section and make a pinned GitHub discussion tutorial or a wiki page for explaining how to contact an instance maintainer.

I think it will be less annoying if it's a GitHub discussion than an issue because we won't feel the pressure of resolving the question directly.

**Comment by dalf** [comment]
Okay !
Then let's go this way.
May I ask you to update the PR and create the new category in the discussions ?

**Pull request merged**